From: Ichthyostega <prg@ichthyostega.de>
Date: Sun, 3 Jun 2018 17:05:36 +0200
Subject: Allow for gzip compressed user manual

Yoshimi provides a menu entry to search for the user manual and display it with the system editor.
Unfortunately, Debian installs most documentation gzip compressed (this is require
by Debian policy and built into the debhelper tools for packaging).

Since xdg-open is able to display a gzip compressed PDF, all we need to do
is also to search for *.pdf.gz in the installed directories.
---
 src/Misc/MiscFuncs.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiscFuncs.cpp b/src/Misc/MiscFuncs.cpp
index e84c6d1..171cc22 100644
--- a/src/Misc/MiscFuncs.cpp
+++ b/src/Misc/MiscFuncs.cpp
@@ -267,7 +267,11 @@ string MiscFuncs::findfile(string path, string filename, string extension)
     fscanf(fp,"%[^\n]", line);
     pclose(fp);
 
-    if (findleafname(line) == filename)
+    string fullName(line);
+    // Extension might contain a dot (like .pdf.gz).
+    // Strip the known extension, but retain trailing dot as marker
+    fullName = fullName.substr(0, fullName.length()-extension.length() + 1);
+    if (findleafname(fullName) == filename)
         return line;
     return "";
 }
